packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -102,6 +103,17 @@ export type ComboBoxProps<T> = {
    * Props to customize dropdown menu behavior.
    */
   dropdownMenuProps?: React.ComponentProps<'div'>;
+  /**
+   * End icon props.
+   */
+  endIconProps?: React.ComponentProps<typeof ComboBoxEndIcon>;
```

Comment:
should we also allow replacing the icon completely?

Reply:
I would rather not because it's combobox.
Users can do InputWithDeco and new Popover if they want something else there.

Reply:
what if i want to show a loading spinner in the combobox (#925)? Popover is not a suitable replacement for all ComboBox functionality.

doesn't necessarily need to be handled in this PR if it's too complicated, but it's good to think about it now, because it affects the API.

Reply:
wont this rotate the custom icon by 180deg when menu opens?

Reply:
Nope, rotation is done with open prop that we are not passing in this case.
Will add story and tests for this, if the approach is ok.

Reply:
hmm yeah this could work. can you add example using ProgressRadial instead of camera?

in the future we can think about how to support multiple icons too; maybe the user wants both arrow and spinner

Reply:
Removed endIcon from this PR.
It requires some css and/or combobox layout changes for it to work properly. Will handle in different PR or after major release.